### ISSUE TITLE
 Unused Sampled Variable xfeat in ProbTri of model.py

### ISSUE THREAD

**Issue by TitansWhale** [bymaths/probabilistic_triangulation#7]
In **model.py**, within ProbTri, there's a sampled variable xfeat that ultimately isn't used. I noticed that you originally intended to feed this variable into the fusion network, but it's commented out. Why was it not used in the end?

```
def forward(self, image, K):
    """
    Args:
        image (B,V,3,H,W)
        K (B,V,3,3)
    """
    B, V, _, image_h, image_w = image.shape
    image = image.view(-1, 3, image_h, image_w)
    K = K.view(-1, 3, 3)
    out_hm, out_reg, feature = self.backbone(image)

    # BV,J,H,W
    heatmap = out_hm
    BV, J, H, W = heatmap.shape
    reg = out_reg.view(BV, J, 2, H * W)

    # ind (BV,J)
    max_val, ind = torch.max(heatmap.view(BV, J, -1), dim=2)
    # ind_xy (BV,J,2)
    ind_y = ind // W
    ind_x = ind % W
    ind_xy = torch.stack([ind_x, ind_y], dim=-1)

    # ind (BV,J,1,1) -> ind (BV,J,2,1) / reg (BV,J,2,HW) / reg_val (BV,J,2)
    reg_val = torch.gather(
        reg, 3, ind.unsqueeze(-1).unsqueeze(-1).expand(-1, -1, 2, -1)
    ).squeeze(3)

    # coord (BV,J,2)
    coord = ind_xy.float() + reg_val
    out_x2d = coord.clone() * image_w / W

    # (BV,1,3,3)@(BV,J,3,1) -> (BV,J,3,1)
    xdir = (
        torch.inverse(K)[:, None] @ eulid_to_homo(coord * image_w / W)[..., None]
    ).squeeze(-1)
    xdir = torch.cat([xdir, max_val[..., None]], dim=-1).view(B, V, J, 4)

    coord[:, :, 0] = (coord[:, :, 0] / (W - 1)) * 2 - 1
    coord[:, :, 1] = (coord[:, :, 1] / (H - 1)) * 2 - 1
    coord = coord.unsqueeze(1)
    xfeat = F.grid_sample(feature, coord, mode="bilinear", align_corners=True)
    xfeat = xfeat.squeeze(2).transpose(1, 2).view(B, V, J, 128)

    # out_x3d = self.fusion(xdir, xfeat)
    out_x3d = self.fusion(xdir)

    return (
        out_x3d,
        out_hm.view(B, V, J, H, W),
        out_reg.view(B, V, J, 2, H, W),
        out_x2d.view(B, V, J, 2),
    )
```

**Comment by bymaths** [comment]
I neglected to use comments.

This part is left over from previous debugging, I will change it later.
I tried to remove xfeat before, and later confirmed that xfeat is indeed beneficial to accuracy.

**Comment by bymaths** [comment]
Thank you for your attention.
This code needs to use the commented part, which was my oversight.

This part is left over from previous debugging, I will change it later.
I tried to remove xfeat before, and later confirmed that xfeat is indeed beneficial to accuracy.
